include/irr/video/IGPUCommandBuffer.h

```diff
@@ -0,0 +1,264 @@
+    asset::VkOffset3D dstOffsets[2];
+};
+struct VkImageResolve
+{
+    asset::IImage::SSubresourceLayers srcSubresource;
+    asset::VkOffset3D srcOffset;
+    asset::IImage::SSubresourceLayers dstSubresource;
+    asset::VkOffset3D dstOffset;
+    asset::VkExtent3D extent;
+};
+struct VkOffset2D
+{
+    int32_t x;
+    int32_t y;
+};
+struct VkExtent2D
+{
+    uint32_t width;
+    uint32_t height;
+};
+struct VkRect2D
+{
+    VkOffset2D    offset;
+    VkExtent2D    extent;
+};
+enum E_STENCIL_FACE_FLAGS : uint32_t
+{
+    ESFF_FRONT_BIT = 0x01,
+    ESFF_BACK_BIT = 0x02,
+    ESFF_FACE_AND_FRONT = 0x03
+};
+enum E_PIPELINE_STAGE_FLAGS : uint32_t
+{
+    EPSF_TOP_OF_PIPE_BIT = 0x00000001,
+    EPSF_DRAW_INDIRECT_BIT = 0x00000002,
+    EPSF_VERTEX_INPUT_BIT = 0x00000004,
+    EPSF_VERTEX_SHADER_BIT = 0x00000008,
+    EPSF_TESSELLATION_CONTROL_SHADER_BIT = 0x00000010,
+    EPSF_TESSELLATION_EVALUATION_SHADER_BIT = 0x00000020,
+    EPSF_GEOMETRY_SHADER_BIT = 0x00000040,
+    EPSF_FRAGMENT_SHADER_BIT = 0x00000080,
+    EPSF_EARLY_FRAGMENT_TESTS_BIT = 0x00000100,
+    EPSF_LATE_FRAGMENT_TESTS_BIT = 0x00000200,
+    EPSF_COLOR_ATTACHMENT_OUTPUT_BIT = 0x00000400,
+    EPSF_COMPUTE_SHADER_BIT = 0x00000800,
+    EPSF_TRANSFER_BIT = 0x00001000,
+    EPSF_BOTTOM_OF_PIPE_BIT = 0x00002000,
+    EPSF_HOST_BIT = 0x00004000,
+    EPSF_ALL_GRAPHICS_BIT = 0x00008000,
+    EPSF_ALL_COMMANDS_BIT = 0x00010000,
+    EPSF_TRANSFORM_FEEDBACK_BIT_EXT = 0x01000000,
+    EPSF_CONDITIONAL_RENDERING_BIT_EXT = 0x00040000,
+    EPSF_RAY_TRACING_SHADER_BIT_KHR = 0x00200000,
+    EPSF_ACCELERATION_STRUCTURE_BUILD_BIT_KHR = 0x02000000,
+    EPSF_SHADING_RATE_IMAGE_BIT_NV = 0x00400000,
+    EPSF_TASK_SHADER_BIT_NV = 0x00080000,
+    EPSF_MESH_SHADER_BIT_NV = 0x00100000,
+    EPSF_FRAGMENT_DENSITY_PROCESS_BIT_EXT = 0x00800000,
+    EPSF_COMMAND_PREPROCESS_BIT_NV = 0x00020000
+};
+enum E_DEPENDENCY_FLAGS : uint32_t
+{
+    EDF_BY_REGION_BIT = 0x01,
+    EDF_DEVICE_GROUP_BIT = 0x04,
+    EDF_VIEW_LOCAL_BIT = 0x02
+};
+enum E_QUERY_CONTROL_FLAGS : uint32_t
+{
+    EQCF_PRECISE_BIT = 0x01
+};
+enum E_QUERY_RESULT_FLAGS : uint32_t
+{
+    EQRF_64_BIT = 0x01,
+    EQRF_WAIT_BIT = 0x02,
+    EQRF_WITH_AVAILABILITY_BIT = 0x04,
+    EQRF_PARTIAL_BIT = 0x08
+};
+
+class IGPUQueue;
+
+class IGPUCommandBuffer : public core::IReferenceCounted
+{
+    friend class IGPUQueue;
+
+public:
+    enum E_RESET_FLAGS : uint32_t
+    {
+        ERF_RELEASE_RESOURCES_BIT = 0x01
+    };
+
+    enum E_STATE : uint32_t
+    {
+        ES_INITIAL,
+        ES_RECORDING,
+        ES_EXECUTABLE,
+        ES_PENDING,
+        ES_INVALID
+    };
+
+    enum E_USAGE : uint32_t
+    {
+        EU_ONE_TIME_SUBMIT_BIT = 0x01,
+        EU_RENDER_PASS_CONTINUE_BIT = 0x02,
+        EU_SIMULTANEOUS_USE_BIT = 0x04
+    };
+
+    enum E_LEVEL : uint32_t
+    {
+        EL_PRIMARY,
+        EL_SECONDARY
+    };
+
+    IGPUCommandBuffer(uint32_t _familyIx) : m_familyIndex(_familyIx)
+    {
+
+    }
```

Comment:
command buffer needs to be created by a pool/driver so no constructor necessary for the interface (at least no public ctor)